pkg/ref/expr.go

```diff
@@ -59,6 +59,14 @@ func Lookup(ctx context.Context, req kclient.Client, out kclient.Object, namespa
 		}
 
 		if validSecrets == nil {
+			switch v := out.(type) {
+			case *corev1.Secret:
+				// Support binding existing secrets with "." in the name, i.e. my-old-app.secret-name
+				if err := r.getSecret(v, namespace, strings.Join(parts, "."), validSecrets); err == nil {
```

Comment:
Do we want to ignore err here?

Reply:
Yes, we do. This `Lookup` function has multiple different uses. This specific case I am adding is for secret public names only. If we get an error while looking for that, we can just ignore it and move on to the next case.

Reply:
Oh got it. Do we want to just ignore the not found error, but in other case, let's say in case of a network error, just return it? It looks like in other cases we only ignore notFound error.

Reply:
Sure, I think that is a good call. I will go add that

Reply:
Thanks, just doing it safely in case there is other type of error.